ethexe/contracts/src/Middleware.sol

```diff
@@ -233,8 +242,33 @@ contract Middleware is IMiddleware, OwnableUpgradeable, ReentrancyGuardTransient
         $.operators.append(msg.sender, 0);
     }
 
+    modifier onlyOperator() {
+        require(_storage().operators.contains(msg.sender), "Operator not registered");
+        _;
+    }
+
+    function registerIdentifier(address identifier) external onlyOperator {
```

Comment:
mb check signature from `identifier`?

Reply:
I think this can be done later in networking on session init.